### PR TITLE
Set up Node Server

### PR DESCRIPTION
**Pull request by lawrencety**
Created a new node server under `src` folder. I set up project as per README but the package-lock.json was still updated for some reason. 

Added the following features, but all untested:

1. Submission Schema
  - Create, read, and get all
  - Routing only for create set up
2. Mailgun integration
  - Still need API key
3. MongoDB server
  - Need to connect to actual server

### CODE REVIEW

File: src/config/main-config.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require('dotenv').config();
+const path = require('path');
+const logger = require('morgan');
+const cors = require('cors')
+const bodyParser = require('body-parser');
+
+module.exports = {
+  init(app, express) {
```
Comment (turnerhayes) [inline src/config/main-config.js]: You can just import `express` in this file, rather than passing it as a parameter

File: src/config/main-config.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    app.use(bodyParser.urlencoded({ extended: true }));
```
Comment (turnerhayes) [inline src/config/main-config.js]: I think we'll probably use JSON payloads rather than URL encoded payloads, so we probably don't need this parser (just bodyParser.json()).

Also, although this isn't a huge deal, it's worth noting that the [bodyParser docs](https://github.com/expressjs/body-parser#express-route-specific) recommend using bodyparsers on a a per-route basis rather than an app-level parser.

File: src/config/main-config.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    app.use(logger('dev'));
+    app.use(express.json());
```
Comment (turnerhayes) [inline src/config/main-config.js:10]: Huh, I wasn't even aware that express had a JSON body parser bundled with it! Sounds like there was [a bit of back-and-forth](https://stackoverflow.com/a/47232318/324399) with it. We can probably do away with the `body-parser` dep in that case.

File: src/config/main-config.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    app.use(cors());
```
Comment (turnerhayes) [inline src/config/main-config.js:11]: I think it's okay to leave this here, but it's probably not necessary since I doubt we'll be serving across origins.

File: src/config/main-config.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    app.use(express.static(path.join(__dirname, '..', 'assets')));
+    app.use((req, res, next) => {
+      res.locals.currentUser = req.user;
```
Comment (turnerhayes) [inline src/config/main-config.js]: Is currentUser going to be used for anything? I don't think we'll have any sort of auth in our app.

File: src/controllers/submissionController.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        statusCode: 200,
+        message: 'Success',
+        data: submissions
+      };
+      res.json(returnData)
+    })
+    .catch((err) => {
+      console.log(err)
+      let returnData = {
+        statusCode: 400,
```
Comment (turnerhayes) [inline src/controllers/submissionController.js]: Probably a `500` rather than a `400`, unless we know the request data is improper

File: src/controllers/submissionController.js [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+        message: 'Bad Request',
+        data: err
+      };
+      res.json(returnData)
```
Comment (turnerhayes) [inline src/controllers/submissionController.js]: We should set `res.status()` so that it doesn't return a `200` on error.

(this and the above comment go equally for the failure paths below, as well)

File: src/routes/static.js [hunk 8]
```diff
@@ -0,0 +1,8 @@
+const express = require("express");
```
Comment (turnerhayes) [inline src/routes/static.js]: Is this file needed?

File: src/server.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
+const app = require('./app');
+const http = require('http');
+const mongoose = require('mongoose');
+
+const uri = ('mongodb://localhost:27017/test' || process.env.mongoSecret);
+mongoose.connect(uri, {useNewUrlParser: true, useFindAndModify: false})
+.then(() => {
```
Comment (turnerhayes) [inline src/server.js:7]: This is a no-op, and can be removed

File: src/server.js [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+})
+.catch((err) => {
+  console.log(err);
```
Comment (turnerhayes) [inline src/server.js]: Should probably use `console.error()` instead, so it goes to stderr instead of stdout

**Review by turnerhayes — COMMENTED** [review]
Looking good! Nothing huge to change, just some minor things.

File: src/routes/static.js (hunk 8 above)
Comment (lawrencety) [reply src/routes/static.js]: Yes, the next router is imported from express
```
const express = require("express");
const router = express.Router();
```

File: src/config/main-config.js (hunk 1 above)
Comment (lawrencety) [reply src/config/main-config.js]: Removed parameter and added dependency.

File: src/config/main-config.js (hunk 2 above)
Comment (lawrencety) [reply src/config/main-config.js]: Agreed, I removed this and left `express.json()`

File: src/config/main-config.js (hunk 3 above)
Comment (lawrencety) [reply src/config/main-config.js:10]: See previous

File: src/config/main-config.js (hunk 4 above)
Comment (lawrencety) [reply src/config/main-config.js:11]: I just used it in case. Not sure if this will affect anything negatively anyways.

File: src/config/main-config.js (hunk 5 above)
Comment (lawrencety) [reply src/config/main-config.js]: Removed. Thanks for catching that.

File: src/controllers/submissionController.js (hunk 6 above)
Comment (lawrencety) [reply src/controllers/submissionController.js]: Understood, we can add a validation layer before to return 400 errors if needed.

File: src/controllers/submissionController.js (hunk 7 above)
Comment (lawrencety) [reply src/controllers/submissionController.js]: Added.

File: src/server.js (hunk 9 above)
Comment (lawrencety) [reply src/server.js:7]: Added a console.log() for now. Can remove if unnecessary.

File: src/server.js (hunk 10 above)
Comment (lawrencety) [reply src/server.js]: Updated.

**Comment by lawrencety** [comment]
Thanks for the notes @turnerhayes. I made the updates as needed and pushed the changes. Let me know if you find anything else.

File: src/routes/static.js (hunk 8 above)
Comment (turnerhayes) [reply src/routes/static.js]: I mean, is `src/routes/static.js` needed? It seems like it just contains a dummy route for the root path.

Comment (lawrencety) [reply src/routes/static.js]: Oh, I see. No, this isn't needed. Removed.

**Comment by lawrencety** [comment]
@turnerhayes I made the final change. Let me know if there is anything else.

**Review by turnerhayes — APPROVED** [review]
Looks good to me; I don't appear to have authority to merge, so this'll have to wait for @anniejstein
